pandas/core/groupby/groupby.py

```diff
@@ -2013,8 +2022,13 @@ def _transform(self, func, *args, engine=None, engine_kwargs=None, **kwargs):
             # result to the whole group. Compute func result
             # and deal with possible broadcasting below.
             # Temporarily set observed for dealing with categoricals.
-            with com.temp_setattr(self, "observed", True):
-                with com.temp_setattr(self, "as_index", True):
+            with com.temp_setattr(self, "as_index", True):
+                if func in ["idxmin", "idxmax"]:
+                    # mypy doesn't recognize func as Literal["idxmin", "idxmax"]
+                    result = self._idxmin_idxmax(
```

Comment:
This looks like it should be equivalent to the `getattr(self, func)` pattern we use below.  any reason we can't avoid special-casing here?  or maybe just trim it to add `kwargs["ignore_unobserved"] = True`?

Reply:
It would be more like

```
func = "_idxmin_idxmax"
kwargs["ignore_unobserved"] = True
```

because `idxmin` and `idxmax` methods don't accept `ignore_unobserved` (it's an internal argument only). I'm not sure if other args/kwargs get in the way here - but I suspect not. If this looks preferable, I can give this a shot.

Reply:
good point, never mind then.  equal levels of special-casing and this is more explicit.  your way is better

Reply:
Does making the post-processing into a method `_wrap_idxmax_idxmin` seem better? We would then branch on the `how` argument.